### DISCUSSION TITLE
Compatibilty issues with Sony camera like ZV-E10 ii in terms of SRT streaming when using Sony's Creator's App.

### DISCUSSION THREAD

**Discussion by CicoZhang** [bluenviron/mediamtx#3919] [Questions]
### Question

There is a previous discussion about this issue: [link](https://github.com/bluenviron/mediamtx/discussions/3387).

As I have tested in the latest version of Sony's Creator's App, **RTMP streaming works with Mediamtx; however, SRT does not work.** As I perceive it, the main reason is that the colon sign":" will be truncated when typed in the URL field, as the screenshot below shows. When typing `srt://192.168.1.7?streamingid=publish:mystream` and saving the settings, the URL becomes `srt://192.168.1.7?streamingid=publish`. I suspect this prevents SRT from working properly.

![Screenshot_20241028_090620](https://github.com/user-attachments/assets/80f4a955-e3d0-4ae9-9ff6-9dd0938d2e30)

Of course, Sony should solve this issue; however, there seems to be no direct channel for reporting this to Sony (Updated: I got the email from Google Play and sent an email to Sony. Update 2: received the email from Sony "Please note that this application is unsupported. We will delete the e-mail you have sent us for inquiry. ").

Is there any workaround?

**Comment by aler9** [comment]
Sorry but we need to use colons (and multiple values) inside streamid because we need to let the server know what we are doing (publishing) and on which path. In SRT, there are no other fields that allow to do that.

Furthermore, the official SRT specification also suggests to use streamid to pass such informations to servers, and it suggests an encoding technique that is based on colons too, see https://github.com/Haivision/srt/blob/master/docs/features/access-control.md

Therefore: contact Sony or avoid the app.
